website/tests/validate.py

```
@@ -0,0 +1,261 @@
+    warnings: list[str] = []
+    missing_pages = sorted(expected - actual)
+    if missing_pages:
+        errors.append(f"missing required pages: {', '.join(missing_pages)}")
+
+    parsed: dict[Path, PageParser] = {}
+    for page in pages:
+        parser = PageParser(page)
+        try:
+            parser.feed(page.read_text(encoding="utf-8"))
+            parser.close()
+        except Exception as exc:  # pragma: no cover
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)